main/signers/lattice/Lattice/index.ts

```diff
@@ -268,21 +262,51 @@ export default class Lattice extends Signer {
 
   async signTransaction (index: number, rawTx: TransactionData, cb: Callback<string>) {
     try {
+      const connection = this.connection as Client
       const compatibility = signerCompatibility(rawTx, this.summary())
       const latticeTx = compatibility.compatible ? { ...rawTx } : londonToLegacy(rawTx)
+          const fwVersion = connection.getFwVersion()
 
       const signedTx = await sign(latticeTx, async tx => {
         const unsignedTx = this.createTransaction(index, rawTx.type, latticeTx.chainId, tx)
-        const signOpts = { currency: 'ETH', data: unsignedTx }
-        const clientSign = promisify((this.connection as Client).sign).bind(this.connection)
+        let signingOptions;
+
+        if (fwVersion && (fwVersion.major > 0 || fwVersion.minor >= 15)) {
+          // @ts-expect-error - Intentionally accessing private method
+          const payload = tx._type ?
+            tx.getMessageToSign(false) :
+            rlp.encode(tx.getMessageToSign(false))
+
+          const to =  tx.to?.toString() ?? undefined
+
+          const callDataDecoder = to 
+            ? await Utils.fetchCalldataDecoder(tx.data, to, unsignedTx.chainId) 
+            : undefined
+
+          const data = {
+            payload,
+            curveType: Constants.SIGNING.CURVES.SECP256K1,
+            hashType: Constants.SIGNING.HASHES.KECCAK256,
+            encodingType: Constants.SIGNING.ENCODINGS.EVM,
+            signerPath: unsignedTx.signerPath,
+            decoder: callDataDecoder?.def
+          };
+          signingOptions = { data, currency: unsignedTx.currency }
+        } else {
+          signingOptions = { currency: 'ETH', data: unsignedTx }
+        }
+
+        const signedTx = await connection.sign(signingOptions)
+        const sig = signedTx?.sig
 
-        const result = await clientSign(signOpts)
-        const sig = result.sig as Signature
+        if (!sig) {
+          throw new Error('Signing failed')
```

Comment:
is there any extra context that can be obtained here to display to the user or for debugging purposes?

Reply:
if its truly not possible then prefer to remove this and it implies that it can happen and could be confusing down the road to people less familiar w. the code